ReportUtilityBills/measure.rb

```diff
@@ -214,29 +214,57 @@ def run(runner, user_arguments)
       get_annual_bills(utility_bills)
 
       # Write/report results
-      write_runperiod_output_results(runner, utility_bills, output_format, output_path, utility_bill_scenario.name)
-      report_runperiod_output_results(runner, utility_bills, utility_bill_scenario.name)
+      report_runperiod_output_results(runner, utility_bills, output_format, output_path, utility_bill_scenario.name)
     end
 
     return true
   end
 
-  def write_runperiod_output_results(runner, utility_bills, output_format, output_path, bill_scenario_name)
+  def report_runperiod_output_results(runner, utility_bills, output_format, output_path, bill_scenario_name)
     segment = utility_bills.keys[0].split(':', 2)[0]
     segment = segment.strip
+
     results_out = []
     results_out << ["#{bill_scenario_name}: Total ($)", utility_bills.values.sum { |bill| bill.annual_total.round(2) }.round(2)]
-    utility_bills.each do |key, bill|
-      new_segment = key.split(':', 2)[0]
+
+    utility_bill_type_str = OpenStudio::toUnderscoreCase("#{bill_scenario_name} Total USD")
+    utility_bill_type_val = utility_bills.values.sum { |bill| bill.annual_total.round(2) }.round(2)
+    runner.registerValue(utility_bill_type_str, utility_bill_type_val)
+    runner.registerInfo("Registering #{utility_bill_type_val} for #{utility_bill_type_str}.")
+
+    utility_bills.each do |fuel_type, bill|
+      new_segment = fuel_type.split(':', 2)[0]
       new_segment = new_segment.strip
       if new_segment != segment
         segment = new_segment
       end
 
-      results_out << ["#{bill_scenario_name}: #{key}: Fixed ($)", bill.annual_fixed_charge.round(2)] if bill.annual_fixed_charge != 0
-      results_out << ["#{bill_scenario_name}: #{key}: Marginal ($)", bill.annual_energy_charge.round(2)] if bill.annual_energy_charge != 0
-      results_out << ["#{bill_scenario_name}: #{key}: PV Credit ($)", bill.annual_production_credit.round(2)] if [FT::Elec].include?(key) && bill.annual_production_credit != 0
-      results_out << ["#{bill_scenario_name}: #{key}: Total ($)", bill.annual_total.round(2)] if bill.annual_total != 0
+      results_out << ["#{bill_scenario_name}: #{fuel_type}: Fixed ($)", bill.annual_fixed_charge.round(2)] if bill.annual_fixed_charge != 0
+      results_out << ["#{bill_scenario_name}: #{fuel_type}: Marginal ($)", bill.annual_energy_charge.round(2)] if bill.annual_energy_charge != 0
+      results_out << ["#{bill_scenario_name}: #{fuel_type}: PV Credit ($)", bill.annual_production_credit.round(2)] if [FT::Elec].include?(fuel_type) && bill.annual_production_credit != 0
+      results_out << ["#{bill_scenario_name}: #{fuel_type}: Total ($)", bill.annual_total.round(2)] if bill.annual_total != 0
+
+      utility_bill_type_str = OpenStudio::toUnderscoreCase("#{bill_scenario_name} #{fuel_type} Fixed USD")
```

Comment:
Can't the code be simplified much further? After `results_out` is populated, loop through each item and do:
```
      name = OpenStudio::toUnderscoreCase(results_out_item[0])
      value = results_out_item[1]
      runner.registerValue(name, value)
      runner.registerInfo("Registering #{value} for #{name}.")
```
Wouldn't that work?